abci/example/kvstore/kvstore.go

```diff
@@ -170,3 +170,18 @@ func (app *Application) Query(reqQuery types.RequestQuery) (resQuery types.Respo
 
 	return resQuery
 }
+
+func (app *Application) PrepareProposal(req types.RequestPrepareProposal) types.ResponsePrepareProposal {
+	return types.ResponsePrepareProposal{
+		BlockData: req.BlockData,
+	}
```

Comment:
According to the ABCI spec, Tendermint MAY return more TXs than fit inside a proposal. The App MUST make sure that the total size of the TXs returned fits within `req.BlockDataSize`.
I am aware you folks are currently using your implementation of `PrepareProposal` (because we were late) but if you plan to converge on upstream ABCI in the upcoming releases, this spot may become a bug.